pkg/customsignaturestore/customsignaturestore.go

```diff
@@ -104,14 +123,106 @@ func (s *Store) String() string {
 	s.lock.Lock()
 	defer s.lock.Unlock()
 
-	if s.customURIs == nil {
-		return "ClusterVersion signatureStores unset, falling back to default stores"
-	} else if len(s.customURIs) == 0 {
+	if s.customStores == nil {
+		return "ClusterVersion signatureStores not set, falling back to default stores"
+	} else if len(s.customStores) == 0 {
 		return "0 ClusterVersion signatureStores"
 	}
-	uris := make([]string, 0, len(s.customURIs))
-	for _, uri := range s.customURIs {
-		uris = append(uris, uri.String())
+	customStores := make([]string, 0, len(s.customStores))
+	for _, customStore := range s.customStores {
+		caName := customStore.CA.Name
+		if caName != "" {
+			customStores = append(customStores, caName)
+		}
+	}
+	return fmt.Sprintf("ClusterVersion signatureStores: %s", strings.Join(customStores, ", "))
+}
+
+// HTTPClient provides a method for generating an HTTP client
+// with the proxy and trust settings, if set in the cluster.
+func (s *Store) HTTPClient(certAuthName string) (*http.Client, error) {
+	transportOption, err := s.getTransport(certAuthName)
+	if err != nil {
+		return nil, err
+	}
+	transportConfig := &transport.Config{Transport: transportOption}
+	transport, err := transport.New(transportConfig)
+	if err != nil {
+		return nil, err
+	}
+	return &http.Client{
+		Transport: transport,
+	}, nil
+}
+
+// getTransport constructs an HTTP transport configuration, including
+// any custom proxy configuration.
+func (s *Store) getTransport(certAuthName string) (*http.Transport, error) {
+	transport := &http.Transport{}
+
+	proxyConfig, err := s.getProxyConfig()
+	if err != nil {
+		return transport, err
+	} else if proxyConfig != nil {
+		proxyFunc := proxyConfig.ProxyFunc()
+		transport.Proxy = func(req *http.Request) (*url.URL, error) {
+			if req == nil {
+				return nil, errors.New("cannot calculate proxy URI for nil request")
+			}
+			return proxyFunc(req.URL)
+		}
+	}
+
+	tlsConfig, err := s.getTLSConfig(certAuthName)
+	if err != nil {
+		return transport, err
+	} else if tlsConfig != nil {
+		transport.TLSClientConfig = tlsConfig
+	}
+	return transport, err
```

Comment:
can be simplified for easier reading

```suggestion
	if err != nil {
		return transport, err
	}
	
	transport.TLSClientConfig = tlsConfig
	return transport, nil
```

Reply:
I will change this to following
```
tlsConfig, err := s.getTLSConfig(certAuthName)
	if err != nil {
		return transport, err
	}

	if tlsConfig != nil {
		transport.TLSClientConfig = tlsConfig
	}
	return transport, nil
```